src/Http/Routing/src/DependencyInjection/RoutingServiceCollectionExtensions.cs

```diff
@@ -98,6 +100,16 @@ public static IServiceCollection AddRouting(this IServiceCollection services)
         // Set RouteHandlerOptions.ThrowOnBadRequest in development
         services.TryAddEnumerable(ServiceDescriptor.Transient<IConfigureOptions<RouteHandlerOptions>, ConfigureRouteHandlerOptions>());
 
+        // JsonOptions
+        if (!TrimmingAppContextSwitches.EnsureJsonTrimmability)
+        {
+#pragma warning disable IL2026 // Suppressed in Microsoft.AspNetCore.Routing.WarningSuppressions.xml
+#pragma warning disable IL3050 // Calling members annotated with 'RequiresDynamicCodeAttribute' may break functionality when AOT compiling.
+            services.TryAddEnumerable(ServiceDescriptor.Singleton<IPostConfigureOptions<JsonOptions>, ConfigureHttpJsonOptions>());
+#pragma warning restore IL3050 // Calling members annotated with 'RequiresDynamicCodeAttribute' may break functionality when AOT compiling.
+#pragma warning restore IL2026 // Suppressed in Microsoft.AspNetCore.Routing.WarningSuppressions.xml
```

Comment:
This shouldn't be part of AddRouting. Maybe make this part of the default web host logic instead?